vars/githubPublishRelease.groovy

```diff
@@ -0,0 +1,133 @@
+    'githubApiUrl',
+    'githubTokenCredentialsId',
+    'githubOrg',
+    'githubRepo',
+    'githubServerUrl',
+    'releaseBodyHeader',
+    'version'
+]
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+        def script = parameters.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        // load default & individual configuration
+        Map config = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .addIfEmpty('githubOrg', script.commonPipelineEnvironment.getGithubOrg())
+            .addIfEmpty('githubRepo', script.commonPipelineEnvironment.getGithubRepo())
+            .addIfEmpty('version', script.commonPipelineEnvironment.getArtifactVersion())
+            .withMandatoryProperty('githubOrg')
+            .withMandatoryProperty('githubRepo')
+            .withMandatoryProperty('githubTokenCredentialsId')
+            .withMandatoryProperty('version')
+            .use()
+
+        new Utils().pushToSWA([step: STEP_NAME], config)
+
+        withCredentials([string(credentialsId: config.githubTokenCredentialsId, variable: 'TOKEN')]) {
+            def releaseBody = config.releaseBodyHeader?"${config.releaseBodyHeader}<br />":''
+            def content = getLastRelease(config, TOKEN)
+            if (config.addClosedIssues)
+                releaseBody += addClosedIssue(config, TOKEN, content.published_at)
+            if (config.addDeltaToLastRelease)
+                releaseBody += addDeltaToLastRelease(config, content.tag_name)
+            postNewRelease(config, TOKEN, releaseBody)
+        }
+    }
+}
+
+Map getLastRelease(config, TOKEN){
+    def result = [:]
+    try {
+        def response = httpRequest "${config.githubApiUrl}/repos/${config.githubOrg}/${config.githubRepo}/releases/latest?access_token=${TOKEN}"
+        result = readJSON text: response.content
+    } catch (e) {
+        echo "[${STEP_NAME}] This is the first release - no previous releases available ${e.getMessage()}"
+        config.addDeltaToLastRelease = false
+    }
+    return result
+}
+
+String addClosedIssue(config, TOKEN, publishedAt){
+    if (config.customFilterExtension) {
+        config.customFilterExtension = "&${config.customFilterExtension}"
+    }
+
+    def publishedAtFilter = publishedAt ? "&since=${publishedAt}": ''
+
+    def response = httpRequest "${config.githubApiUrl}/repos/${config.githubOrg}/${config.githubRepo}/issues?access_token=${TOKEN}&per_page=100&state=closed&direction=asc${publishedAtFilter}${config.customFilterExtension}"
```

Comment:
[nit] what about checking `response.status`?

Reply:
@marcusholl code is updated, please have another look ...